src/NServiceBus.AcceptanceTests/Core/Routing/When_replying_with_int_or_enum.cs

```diff
@@ -0,0 +1,77 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.Routing
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+
+    public class When_replying_with_int_or_enum : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_throw_a_good_exception()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(c => c.When(b => b.SendLocal(new MyRequest())))
+                .Done(c => c.GotTheRequest)
+                .Run();
+
+            Assert.NotNull(context.ExceptionFromReply);
+            StringAssert.Contains("Could not find metadata for", context.ExceptionFromReply.Message, "");
```

Comment:
Does the specific exception text actually matter? Seems like the other asserts are good enough to prevent a regression.